### ISSUE TITLE
No DAG or Tools displayed for workflow cwltool says is valid

### ISSUE THREAD

**Issue by wshands** [dockstore/dockstore#3349]
DAG and Tools tab says "No Records Found" and DAG tab says "DAG cannot be created because some required tools are missing from Github repo."  for Workflow https://dockstore.org/workflows/github.com/hubmapconsortium/sc-atac-seq-pipeline/HuBMAP_scATAC-seq_Pipeline:develop?tab=dag

workflow contains SubworkflowFeatureRequirement {} :
```
$ grep SubworkflowFeatureRequirement *.cwl
create_snap_and_analyze.cwl:  SubworkflowFeatureRequirement: {}
```

cwltool says it is valid:

```
$ cwltool --validate create_snap_and_analyze.cwl 
INFO /home/ubuntu/.local/bin/cwltool 1.0.20191022103248
INFO Resolved 'create_snap_and_analyze.cwl' to 'file:///mnt/gitroot/sc-atac-seq-pipeline/create_snap_and_analyze.cwl'
create_snap_and_analyze.cwl is valid CWL.
```
However Dockstore webservice generate exception at io/dockstore/webservice/languages/CWLHandler.java:270
with message 

```
java.lang.IllegalStateException: Not a JSON Array: {"ScatterFeatureRequirement":{},"SubworkflowFeatureRequirement":{}}
```

**To Reproduce**
Go to https://dockstore.org/workflows/github.com/hubmapconsortium/sc-atac-seq-pipeline/HuBMAP_scATAC-seq_Pipeline:develop?tab=dag
Go to https://dockstore.org/workflows/github.com/hubmapconsortium/sc-atac-seq-pipeline/HuBMAP_scATAC-seq_Pipeline:develop?tab=tools

**Expected behavior**
Should see tools and DAG properly displayed

**Screenshots**
![Screen Shot 2020-03-25 at 8 36 47 PM](https://user-images.githubusercontent.com/3650921/77607703-6e418380-6ed8-11ea-9b81-7b5f64699042.png)
![Screen Shot 2020-03-25 at 8 36 40 PM](https://user-images.githubusercontent.com/3650921/77607706-6f72b080-6ed8-11ea-97f0-6a6f18ede731.png)

**Desktop (please complete the following information):**
 - OS: iOS
 - Browser Google Chrome
 - Version Version 80.0.3987.132 (Official Build) (64-bit)

**Additional context**
 - UI version: 2.5.1
 - Webservice version: 1.8.1
 - host location: dockstore.org

┆Issue is synchronized with this [Jira Bug](https://ucsc-cgl.atlassian.net/browse/DOCK-1281)
┆Fix Versions: Dockstore 1.9.X
┆Issue Number: DOCK-1281
┆Sprint: Sprint 46- Unicorn Fish
┆Issue Type: Bug

**Comment by agduncan94** [comment]
This may be one of those issues with our CWL avro library not knowing there can be arrays for those fields.

**Comment by wshands** [comment]
Note that the Common Workflow Language User Guide also shows this example:
https://www.commonwl.org/user_guide/22-nested-workflows/index.html

**Comment by wshands** [comment]
- class: SubworkflowFeatureRequirement works in Dockstore
instead of SubworkflowFeatureRequirement: {}
which Dockstore flagged as invalid JSON

**Comment by wshands** [comment]
Also a problem for https://github.com/dockstore-testing/multi-step-cwl

**Comment by NatalieEO** [comment]
The problem described cannot be reproduced using the workflow described. When I tried fixing this the first time, I used the workflow https://github.com/dockstore-testing/multi-step-cwl that Walt mentioned in the comment. I also could not reproduce the original issue, but fixed another bug I noticed at the time.

However, after looking at it again, I wasn't using the correct file as the main descriptor. I was using `sub_workflow_align_and_metrics.cwl` which does include a subworkflow requirement https://github.com/NatalieEO/multi-step-cwl/blob/f2c1fe50599aa130925bafa50d1498e7aedbdeb5/version_1_2/sub_workflow_align_and_metrics.cwl#L64. Registering the workflow using this path doesn't cause the error described above.

After switching to the right descriptor `uber_workflow_align_and_metrics.cwl`, I still can't reproduce the bug described but there are other problems anyway...Created #3857. #3352 is also an issue for this workflow.
